pandas/tests/test_algos.py

```diff
@@ -230,21 +230,11 @@ def test_factorize_nan(self):
         key = np.array([1, 2, 1, np.nan], dtype="O")
         rizer = ht.ObjectFactorizer(len(key))
         for na_sentinel in (-1, 20):
-            ids = rizer.factorize(key, sort=True, na_sentinel=na_sentinel)
-            expected = np.array([0, 1, 0, na_sentinel], dtype="int32")
```

Comment:
Just to make sure I'm understanding - the dtype in main currently is wrong, but the old test never actually checked it. Is that right?

Reply:
Correct. Honestly, I am not sure what the old test was supposed to test at all.

We get int32 on 32-bit systems, so I disabled the dtype check, it's not relevant here anyway